uc-catalog-cloning/clonecatalog.py

```diff
@@ -0,0 +1,448 @@
+from py4j.protocol import Py4JError
+from typing import Dict, Optional, List
+import re
+import logging
+
+# Set up logger to capture logs and store them in a file
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+file_handler = logging.FileHandler(filename="logs.log", mode="w")
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+# Attempt to retrieve the Databricks environment's API URL
+# This block checks whether the code is running inside a Databricks notebook.
+try:
+    dbutils.notebook.entry_point.getDbutils().notebook().getContext().apiUrl().get()
```

Comment:
this is an internal api and you don't need this call.

why do you need it?

Reply:
I put it to automatically determine if the code is run from Databricks or locally

Reply:
more security conscious way is checking the `DATABRICKS_RUNTIME_VERSION` - see https://github.com/databricks/databricks-sdk-py/blob/193c3fbc255262de506b9bb927c87392c2c26720/databricks/sdk/core.py#L102-L123

Reply:
Sure, I removed 
try:
    dbutils.notebook.entry_point.getDbutils().notebook().getContext().apiUrl().get()